src/OrchardCore.Modules/OrchardCore.Media/Handlers/MediaFieldIndexHandler.cs

```diff
@@ -0,0 +1,62 @@
+using System.Linq;
+using System.Threading.Tasks;
+using OrchardCore.Indexing;
+using OrchardCore.Media.Fields;
+using OrchardCore.Media.Settings;
+using UglyToad.PdfPig;
+
+namespace OrchardCore.Media.Handlers
+{
+    public class MediaFieldIndexHandler : ContentFieldIndexHandler<MediaField>
+    {
+        private readonly IMediaFileStore _mediaFileStore;
+
+        public MediaFieldIndexHandler(IMediaFileStore mediaFileStore)
+        {
+            _mediaFileStore = mediaFileStore;
+        }
+
+        public async override Task BuildIndexAsync(MediaField field, BuildFieldIndexContext context)
+        {
+            var options = context.Settings.ToOptions();
+            var settings = context.ContentPartFieldDefinition.GetSettings<MediaFieldSettings>();
+
+            if (field.Paths.Length > 0)
+            {
+                if (settings.AllowMediaText)
+                {
+                    foreach (var key in context.Keys)
+                    {
+                        foreach (var mediaText in field.MediaTexts)
+                        {
+                            context.DocumentIndex.Set(key, mediaText, options);
+                        }
+                    }
+                }
+
+                foreach (var path in field.Paths.Where(path => path.EndsWith(".pdf")))
+                {
+                    using var fileStream = await _mediaFileStore.GetFileStreamAsync(path);
+                    if (fileStream != null)
+                    {
+                        using var document = PdfDocument.Open(fileStream);
+                        foreach (var page in document.GetPages())
+                        {
+                            foreach (var key in context.Keys)
+                            {
+                                context.DocumentIndex.Set(key, page.Text, options);
```

Comment:
This should always be analyzed.

Reply:
Why? We have the same way of configuring indexing for all fields, so whether to analyze and/or store is coming from the field settings.

Reply:
Is it relevant to index/store the entire PDF page to search for a term in it?

Reply:
I don't see why it wouldn't be.

Reply:
key + ".PdfText"

Reply:
While I changed the key I'm still not sure about your initial remark.

Reply:
The only advantage I can see is if we allow to "store" the text of a PDF it won't need to be parsed again by PdfPig. Though, what you want is to be able to find specific terms in these PDF so, you want to analyze/tokenize that text more than just storing it. Storing is generally used on technical values like a GUID or a SKU. These makes sense to be only stored as they are generally unique.

Reply:
Hmm, I see. I could hard-code `options |= DocumentIndexOptions.Analyze;` here but wouldn't it be awkward that it doesn't fully honor the configuration selected on the admin?

Reply:
It would still honor the configuration for the MediaText. We do this also for the TitlePart ... I think what we eventually will need is something to disable these configs per field if needed. Here for example we would need 2 different configs since we index 2 different values. These indexes configurations needs to be enhanced to allow doing this kind of thing. It's been thinked only for fields that indexes a single value.

Reply:
OK, then, added it.